### PR TITLE
fix(module:carousel): support dynamic change of nz-carousel-content

### PR DESCRIPTION
**Pull request by vthinkxie**
Closes NG-ZORRO/ng-zorro-antd#56

### CODE REVIEW

File: src/components/carousel/nz-carousel.component.ts [hunk 1]
```diff
@@ -83,14 +96,18 @@ export class NzCarouselComponent implements AfterViewInit, OnDestroy {
       if (this.nzAutoPlay) {
         this.createInterval();
       }
-      this._renderer.setStyle(this.slickList.nativeElement, 'height', `${this.hostElement.nativeElement.offsetHeight}px`);
+
       if (this.nzVertical) {
+        this._renderer.setStyle(this.slickList.nativeElement, 'height', `${this.hostElement.nativeElement.offsetHeight}px`);
```
Comment (giscafer) [inline src/components/carousel/nz-carousel.component.ts]: One line of code is missing here.
`this._renderer.removeStyle(this.slickList.nativeElement, 'height');`

if not，in vertical mode, the slickList's height will be setting to  0 .

Comment (vthinkxie) [inline src/components/carousel/nz-carousel.component.ts]: slickList's height is defined when ngAfterViewInit trigger, this height won't change when vertical mode. If remove it style here, the height will be multiple here.

Comment (giscafer) [inline src/components/carousel/nz-carousel.component.ts]: @vthinkxie if not, the height is 0,  `nz-demo-carousel-vertical` demo can't display normally

you should test it by modify the code:


*NzDemoCarouselVerticalComponent*

```
export class NzDemoCarouselVerticalComponent implements OnInit {
 array = [  ]; // try dynamic change the array
  
    constructor() {
    }
  
    ngOnInit() {
  setTimeout(()=>{
      this.array = [ 1, 2, 3, 4 ];
    },500)
    }
  }

```

Comment (vthinkxie) [inline src/components/carousel/nz-carousel.component.ts]: you are right. I will update this code.

**Pull request merged**